### PR TITLE
Splash screen update

### PR DESCRIPTION
**Pull request by ComicSAS**
### Description
- Updated splash screen logic & animation
- Fixed ServerSetup back navigation bug

### Checklist
- [x] I have ensured this PR does not contain any breaking changes for existing functionality;
- [x] I have added automated tests that cover the new behavior and removed obsolete tests and code;

### Tested On
- [ ] Emulator
- [x] Real device

### CODE REVIEW

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt [hunk 1]
```diff
@@ -47,12 +52,27 @@ class AiStableDiffusionActivity : AppCompatActivity() {
         debugLog("Storage permission is ${result}.")
     }
 
+    @SuppressLint("Recycle")
```
Comment (ShiftHackZ) [inline presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt:55]: What is this ?

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionViewModel.kt [hunk 2]
```diff
@@ -27,6 +29,9 @@ class AiStableDiffusionViewModel(
 
     override val effectDispatcher = dispatchersProvider.immediate
 
+    private val _isShowSplash = MutableStateFlow(true)
+    val isShowSplash = _isShowSplash.asStateFlow()
```
Comment (ShiftHackZ) [inline presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionViewModel.kt:33]: Why introduce another stateflow variable in MVI viewmodel?
Please add this as a variable to the MviState of this viewmodel.

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionViewModel.kt [hunk 3]
```diff
@@ -59,4 +64,8 @@ class AiStableDiffusionViewModel(
             homeRouter.navigateToRoute(intent.route)
         }
     }
+
+    fun hideSplash() {
+        _isShowSplash.value = false
+    }
```
Comment (ShiftHackZ) [inline presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionViewModel.kt:70]: Do not expose this viewModel method. Please create another MviIntent for this specific action.

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt [hunk 4]
```diff
@@ -47,12 +52,27 @@ class AiStableDiffusionActivity : AppCompatActivity() {
...
     override fun onCreate(savedInstanceState: Bundle?) {
-        installSplashScreen()
         super.onCreate(savedInstanceState)
         actionBar?.hide()
-        PermissionUtil.checkNotificationPermission(this, notificationPermission::launch)
-        PermissionUtil.checkStoragePermission(this, storagePermission::launch)
+        val splashScreen = installSplashScreen()
```
Comment (ShiftHackZ) [inline presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt:59]: I suppose this variable should be called before the `super.onCreate()` call.
Did you read the splash screen library [documentation](https://developer.android.com/develop/ui/views/launch/splash-screen/migrate#migrate)?

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt (hunk 1 above)
Comment (ComicSAS) [reply presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt:55]: Removed.

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionViewModel.kt (hunk 2 above)
Comment (ComicSAS) [reply presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionViewModel.kt:33]: Resolved.

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionViewModel.kt (hunk 3 above)
Comment (ComicSAS) [reply presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionViewModel.kt:70]: Resolved.

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt (hunk 4 above)
Comment (ComicSAS) [reply presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt:59]: Resolved.

**Review by ShiftHackZ — APPROVED** [review]

**Pull request merged**